server/src/main/java/io/crate/execution/ddl/tables/AlterTableClient.java

```diff
@@ -229,36 +226,29 @@ private CompletableFuture<Long> setSettings(BoundAlterTable analysis) {
 
     private CompletableFuture<Long> resize(BoundAlterTable analysis) {
         final TableInfo table = analysis.table();
-        final boolean isPartitioned = analysis.isPartitioned();
-        String sourceIndexName;
-        String sourceIndexAlias;
-        if (isPartitioned) {
-            PartitionName partitionName = analysis.partitionName();
-            assert partitionName != null
-                : "Resizing operations for partitioned tables are only supported at partition level";
```

Comment:
Just to be sure: is this documented? 
We don't treat this a bug do we? 
If want to change this, maybe worth adding a TODO

Reply:
turns out this is expected. If no PARTITION clause is present it behaves like "TABLE ONLY" in that only the template is updated, and new partitions will have the new shard size but existing partitions are left untouched.